.github/workflows/test-all-versions.yml

```diff
@@ -13,8 +13,35 @@ on:
         type: string
 
 jobs:
+  build:
```

Comment:
note for reviewer: this is the build step that is going to be cached. `github.run_number` is unique for the PR so each time the workflow is run this tasks overwrites the artifact saving space. `retention-days` is set to 1 so the artifact will be removed the next day after the last run of this workflow.